test/test_db.py

```diff
@@ -39,8 +39,8 @@ def test_concurrency():
     service_instance = Mock()
 
     def inject(worker_ctx):
```

Comment:
this is accessing container internals directly. would be good if that wasn't required for extension authors. could this not e.g. just call `spawn_worker` or similar?